x-pack/plugins/observability/server/services/slo/find_slo_groups.ts

```diff
@@ -66,12 +67,25 @@ export class FindSLOGroups {
   public async execute(params: FindSLOGroupsParams): Promise<FindSLOGroupsResponse> {
     const pagination = toPagination(params);
     const groupBy = params.groupBy;
+    const kqlQuery = params.kqlQuery ?? '';
+    const filters = params.filters ?? '';
+    let parsedFilters: any = {};
+
+    try {
+      parsedFilters = JSON.parse(filters);
+    } catch (e) {
+      console.error(`Failed to parse filters: ${e.message}`);
+    }
     const response = await this.esClient.search<unknown, GroupAggregationsResponse>({
       index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
       size: 0,
       query: {
         bool: {
-          filter: [{ term: { spaceId: this.spaceId } }],
+          filter: [
+            { term: { spaceId: this.spaceId } },
+            getElastichsearchQueryOrThrow(kqlQuery),
+            ...(parsedFilters.filter ?? []),
+          ],
         },
       },
       body: {
```

Comment:
Body is deprecated, just use "aggs" directly.

Reply:
Ok I tested your suggestion and it works. Since CI is now green, I will apply this suggestion in a next PR along with some other small changes we discussed (adding a logger, showing number of groups etc)